components/script/dom/element.rs

```diff
@@ -1071,16 +1081,19 @@ impl<'a> ElementMethods for JSRef<'a, Element> {
 
     // http://dom.spec.whatwg.org/#dom-element-removeattribute
     fn RemoveAttribute(self, name: DOMString) {
-        let name = self.parsed_name(name);
-        self.remove_attribute(ns!(""), &name)
+        let name = &Atom::from_slice(&self.parsed_name(name));
+        self.remove_attribute(|attr| attr.name() == name);
     }
 
     // http://dom.spec.whatwg.org/#dom-element-removeattributens
     fn RemoveAttributeNS(self,
                          namespace: Option<DOMString>,
                          localname: DOMString) {
-        let namespace = namespace::from_domstring(namespace);
-        self.remove_attribute(namespace, &localname)
+        let namespace = &namespace::from_domstring(namespace);
+        let localname = &Atom::from_slice(&localname);
+        self.remove_attribute(|attr| {
```

Comment:
I think a `remove_attribute_ns(ns, name)` would be useful to have in these cases (internally just delegates to `remove_attribute()` with the correct filter)